crates/l1tx/src/filter/mod.rs

```diff
@@ -79,6 +81,21 @@ fn parse_checkpoint_envelopes<'a>(
     })
 }
 
+/// Parse transaction and filter out any deposits that have been spent.
+fn parse_deposit_spends<'a>(
+    tx: &'a Transaction,
+    filter_conf: &'a TxFilterConfig,
+) -> impl Iterator<Item = DepositSpendInfo> + 'a {
+    tx.input.iter().filter_map(|txin| {
+        filter_conf
+            .expected_outpoints
+            .binary_search_by_key(&txin.previous_output, |config| config.output.outpoint())
```

Comment:
Leaving a note here on this since I would expect that it'd be reworked again as well.